src/parser/HLSTree.cpp

```diff
@@ -102,8 +103,8 @@ int HLSTree::processEncryption(std::string baseUrl, std::map<std::string, std::s
   if (map["METHOD"] == "AES-128" && !map["URI"].empty())
   {
     current_pssh_ = map["URI"];
-    if (current_pssh_[0] != '/' && current_pssh_.find("://") == std::string::npos)
-      current_pssh_ = baseUrl + current_pssh_;
+    if (!URL::IsValidUrl(current_pssh_))
+      current_pssh_ = URL::Join(baseUrl, current_pssh_);
```

Comment:
not sure to understand exaclty
you mean that in this specific case is not needed call `BuildDownloadUrl`
or you mean delete `BuildDownloadUrl` method?
if this last case, delete this helper method dont seems good, since it is called from different points

Reply:
now that i see for the line 106 `if (!URL::IsValidUrl(current_pssh_))`
the url should come from the manifest need really to check validation?
do you think we can replace it as:
 `if (!URL::IsUrlAbsolute(current_pssh_))`

Reply:
removed -- never mind my overisght

Reply:
yes, just that specific case, with the code above you'll always get a full url right?

Reply:
yes you have right i will remove it